base/io.jl

```diff
@@ -448,7 +456,32 @@ function readuntil(s::IO, t::AbstractString)
 end
 
 readline() = readline(STDIN)
-readline(s::IO) = readuntil(s, '\n')
+
+function readline(s::IO, chomp = false; nl2lf=false)
```

Comment:
I think it'd be better (and more performant) to actually make `readuntil` be able to accept a `Vector{Char}`. The problem here is that `readuntil` itself is creating an IOBuffer and `take!`ing the output at the end, and this approach layers yet another IOBuffer on top of that.

The difficulty will be how you deal with `\r\n`, since it it's doing a special case here.